### PR TITLE
DLG System is ready for implementation

### PR DESCRIPTION
**Pull request by DrunkReaperMatt**
Plugin ready to merge

NB.
- **New** Blueprint Class from Rosetta Character
- **New** Game mode Blueprint 
- **New** Added Blueprint properties for dialogue trees files

### CODE REVIEW

File: Source/Rosetta/RosettaCharacter.cpp [hunk 1]
```diff
@@ -13,6 +13,10 @@
 #include <Kismet/GameplayStatics.h>
 #include "Engine.h"
 
+#include "DlgContext.h"
+#include "DlgManager.h"
+#include "NonPlayerCharacter.h"
```
Comment (karakune) [inline Source/Rosetta/RosettaCharacter.cpp]: est-ce que ce include est nécessaire?

File: Source/Rosetta/RosettaCharacter.cpp [hunk 2]
```diff
@@ -194,6 +198,8 @@ void ARosettaCharacter::Interact()
 	if (InteractableHit)
 	{
 		InteractableHit->Interact();
+
+		//StartDialogue(Cast<ANonPlayerCharacter>(InteractableHit)->dialogueData, Cast<ANonPlayerCharacter>(InteractableHit));
```
Comment (karakune) [inline Source/Rosetta/RosettaCharacter.cpp]: À retirer

File: Source/Rosetta/RosettaCharacter.h [hunk 3]
```diff
@@ -101,4 +102,37 @@ class ARosettaCharacter : public ACharacter
...
 	void UpdateDictionary(FString OriginalWord, FString NewTranslation);
 	TMap<FString, FString> GetDictionary();
-};
+
+
+	/// DLG Dialogue
+public:
+
+
+	// 	UPROPERTY(EditAnywhere, BlueprintReadOnly, Category = DlgData, meta = (AllowPrivateAccess = "true"))
```
Comment (karakune) [inline Source/Rosetta/RosettaCharacter.h]: À retirer?

File: Source/Rosetta/NonPlayerCharacter.h [hunk 4]
```diff
@@ -4,23 +4,42 @@
...
+
+
 
 	//UFUNCTION(BlueprintGetter)
 	virtual FText GetActionDescription() override;
+
+	class ARosettaCharacter* player;
+
+public:
+	virtual void Interact()override;
```
Comment (karakune) [inline Source/Rosetta/NonPlayerCharacter.h]: add space after ) for clarity

File: Source/Rosetta/NonPlayerCharacter.cpp [hunk 5]
```diff
@@ -5,23 +5,42 @@
...
+
 ANonPlayerCharacter::ANonPlayerCharacter()
 {
 	ConstructorHelpers::FClassFinder<UUserWidget> DWClassFinder(TEXT("/Game/UI/DialogueWidget"));
 	DWClass = DWClassFinder.Class;
 }
 
-void ANonPlayerCharacter::Interact()
-{
+/// Old Code from widget
```
Comment (karakune) [inline Source/Rosetta/NonPlayerCharacter.cpp]: delete if not-necessary

File: Source/Rosetta/DialogueWidget.cpp [hunk 6]
```diff
@@ -90,50 +89,52 @@ void UDialogueWidget::GenerateWordWidgets(FString Sentence)
...
-			}
-			break;
-		case 6:
-			UDialogueWidget::CloseWidget();
-			break;
-	}
+	Player->SelectDialogueOption(DialogueStep);
+	UpdateOption(DialogueStep);
+
+	//	DialogueStep++;
```
Comment (karakune) [inline Source/Rosetta/DialogueWidget.cpp]: delete if not-necessary

**Review by karakune — CHANGES_REQUESTED** [review]

File: Source/Rosetta/RosettaCharacter.cpp (hunk 1 above)
Comment (DrunkReaperMatt) [reply Source/Rosetta/RosettaCharacter.cpp]: npc.h est superflu

removed

File: Source/Rosetta/RosettaCharacter.h (hunk 3 above)
Comment (DrunkReaperMatt) [reply Source/Rosetta/RosettaCharacter.h]: est retirer

File: Source/Rosetta/DialogueWidget.cpp [hunk 7]
```diff
@@ -28,23 +28,23 @@ bool UDialogueWidget::Initialize()
 	Super::Initialize();
 	OkayButton->OnClicked.AddDynamic(this, &UDialogueWidget::OkayPressed);
 	ResponseInput->OnTextCommitted.AddDynamic(this, &UDialogueWidget::DeactivateResponse);
-	
 	UDialogueWidget::Setup();
 
-	UDialogueWidget::GenerateWordWidgets(DialogueIntroLine);
+	UpdateOption(DialogueStep);
+	Player->widget = this;
```
Comment (karakune) [inline Source/Rosetta/DialogueWidget.cpp:34]: renommer widget pour quelque chose de plus explicite

**Review by karakune — APPROVED** [review]

**Pull request merged**